presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousBatchTaskStatusFetcher.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.remotetask;
+
+import com.facebook.airlift.concurrent.SetThreadName;
+import com.facebook.airlift.http.client.HttpClient;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.execution.StateMachine;
+import com.facebook.presto.execution.TaskId;
+import com.facebook.presto.execution.TaskStatus;
+import com.facebook.presto.server.smile.BaseResponse;
+import com.facebook.presto.server.smile.Codec;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.Duration;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Consumer;
+
+import static java.util.Objects.requireNonNull;
+
+public class ContinuousBatchTaskStatusFetcher
+{
+    public class Task {
+        TaskId taskId;
+        StateMachine<TaskStatus> taskStatus;
+    }
+
+    // TaskId -> WorkerId (String(URI(TaskStatus.getSelf() (get worker id)))
+    private final ConcurrentHashMap<TaskId, String> idWorkerMap;
+    // String(URI(TaskStatus.getSelf() (get worker id))) -> WorkerTaskStatusFetcher (List<Task>)
+    private final ConcurrentHashMap<String, WorkerTaskStatusFetcher> workerTaskMap;
+
+    private static final Logger log = Logger.get(ContinuousBatchTaskStatusFetcher.class);
+
+    private final Consumer<Throwable> onFail;
+    private final Codec<Map<TaskId, TaskStatus>> taskListStatusCodec;
+    private final Duration refreshMaxWait;
+    private final Executor executor;
+    private final HttpClient httpClient;
+    private final Duration maxErrorDuration;
+    private final ScheduledExecutorService errorScheduledExecutor;
+    private final RemoteTaskStats stats;
+    private final boolean isBinaryTransportEnabled;
+
+    @GuardedBy("this")
+    private ListenableFuture<BaseResponse<TaskStatus>> future;
+
+    public ContinuousBatchTaskStatusFetcher(
+            Consumer<Throwable> onFail,
+            Codec<Map<TaskId, TaskStatus>> taskListStatusCodec,
+            Duration refreshMaxWait,
+            Executor executor,
+            HttpClient httpClient,
+            Duration maxErrorDuration,
```

Comment:
remove this and initialize it with below in constructor:
config = QueryManagerConfig
`maxErrorDuration = config.getRemoteTaskMaxErrorDuration()`